### DISCUSSION TITLE
WinUI3 : Can a control/window creation through C++ fail?

### DISCUSSION THREAD

**Discussion by harshith187** [microsoft/WindowsAppSDK#4019] [Q&A] [answered]
Hi,

I'm working on WinUI3 desktop app in CPP. I'm creating the UI programmatically without using XAML like this

Window win = Window();

Can this call to window constructor fail?

I went through the [documentation](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.window.-ctor?view=windows-app-sdk-1.3#microsoft-ui-xaml-window-ctor), but it did not mention anything about this function failing.

I also posted this question in MS Q&A - Ref (https://learn.microsoft.com/en-gb/answers/questions/1324279/winui3-can-a-control-window-creation-through-c-fai), where I got to know that this function is newer meant to fail. But in the hypothetical scenario like if no memory is available in system when I create this widget, will the widget creation fail? how is this handled?

Thank you

**Comment by DarranRowe** [comment, marked as answer]
Generally, C++/WinRT will use C++ exceptions. So it is usually nice to have a catch block for winrt::hresult_error to handle this.

**Reply by harshith187** [reply]
Thank you

**Comment by jaigak** [comment]
> I'm creating the UI programmatically without using XAML like this

@harshith187 Are you initialising the XAML framework correctly? You have to call Application::Start.

**Reply by harshith187** [reply]
Yes, I have initialized the threading model to single threaded apartment. Initialised bootstrap and started the event loop with application start
